tests/components/bsblan/test_water_heater.py

```diff
@@ -0,0 +1,208 @@
+
+    freezer.tick(timedelta(minutes=1))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(ENTITY_ID)
+    assert state.attributes.get("temperature") == 10
+
+
+@pytest.mark.parametrize(
+    ("mode", "bsblan_mode"),
+    [
+        (STATE_ECO, "Eco"),
+        (STATE_OFF, "Off"),
+        (STATE_ON, "On"),
+    ],
+)
+async def test_set_operation_mode(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    mode: str,
+    bsblan_mode: str,
+) -> None:
+    """Test setting operation mode."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    await hass.services.async_call(
+        domain=WATER_HEATER_DOMAIN,
+        service=SERVICE_SET_OPERATION_MODE,
+        service_data={
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_OPERATION_MODE: mode,
+        },
+        blocking=True,
+    )
+
+    mock_bsblan.set_hot_water.assert_called_once_with(operating_mode=bsblan_mode)
+
+
+async def test_set_invalid_operation_mode(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting invalid operation mode."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    with pytest.raises(
+        HomeAssistantError,
+        match=r"Operation mode invalid_mode is not valid for water_heater\.bsb_lan\. Valid operation modes are: eco, off, on",
+    ):
+        await hass.services.async_call(
+            domain=WATER_HEATER_DOMAIN,
+            service=SERVICE_SET_OPERATION_MODE,
+            service_data={
+                ATTR_ENTITY_ID: ENTITY_ID,
+                ATTR_OPERATION_MODE: "invalid_mode",
+            },
+            blocking=True,
+        )
+
+
+async def test_set_temperature(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting temperature."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    await hass.services.async_call(
+        domain=WATER_HEATER_DOMAIN,
+        service=SERVICE_SET_TEMPERATURE,
+        service_data={
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_TEMPERATURE: 50,
+        },
+        blocking=True,
+    )
+
+    mock_bsblan.set_hot_water.assert_called_once_with(nominal_setpoint=50)
+
+
+async def test_set_temperature_failure(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting temperature with API failure."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    mock_bsblan.set_hot_water.side_effect = BSBLANError("Test error")
+
+    with pytest.raises(HomeAssistantError, match="set_temperature_error"):
```

Comment:
Wait how does this work? Why isn't this translated? Do you have translation strings for these?

Reply:
I guess not. Will set them. I am away currently

Reply:
Also in climate I need to set the I see